frontend/README.md

```diff
@@ -0,0 +1,83 @@
+# Stable Diffusion Web UI
+
+Demo at https://peaceful-otter-7a427f.netlify.app/ (not connected to back end)
+
+much of this readme is just notes for myself during dev work
+
+## Test and Build
+
+from `frontend/`:
+
+-   `yarn dev` runs `tsc-watch`, which runs `vite build` on successful `tsc` transpilation
```

Comment:
Any reason not to just use the [vite dev server](https://vitejs.dev/config/server-options.html)? 

It can be configured to [proxy requests](https://vitejs.dev/config/server-options.html#server-proxy) to the local dream server in order to get around CORS issues (it even works with websockets)

By manually watching like this, you're missing out on a lot of time-saving performance benefits of vite's dev server.

Reply:
I had a few occasions where the vite dev server  it did not catch certain TS errors in code and there was no error in the terminal  and no error in the site's JS console. I had to kill & restart the `vite dev` process at which point it caught the errors. I'm not sure why that happened but doing it this way catches all TS errors though as you said much slower. Very possible I was doing something wrong.

Reply:
Vite [doesn't do type-checking](https://vitejs.dev/guide/features.html#typescript) as a [side-effect of using esbuild](https://esbuild.github.io/content-types/#no-type-system).

I usually just rely on my IDE to report type errors while developing, and run `tsc` as a check at the end before pushing changes (and during CI/CD/etc) as if it were a linting or testing step.

Reply:
Ahh yep that was the issue. I've set up my IDE to handle everything now. Thanks.